### ISSUE TITLE
Support input sourcemaps in parse

### ISSUE THREAD

**Issue by mariusGundersen** [benjamn/recast#115]
When parsing a js file with sourcemaps the locations in the AST output should point to the original file locations, not the parsed file locations.

**Comment by caridy** [comment]
related to https://github.com/benjamn/recast/pull/146

**Comment by gdborton** [comment]
@benjamn Is this possible yet?  I'm trying to transform then concatenate js files and source maps into a single bundle. This works well if I pass around the ast from `recast.parse(source)`, but doesn't seem to work if I try to reuse the ast generated from babel `babel.transform(source, {
ast: true })`.

I think ideally I'd be able to call
`recast.parse(babelTranspiledSource, { inputSourceMap: transpiledMap })` and then carry on my merry way.

Basically I need to be able to call a babel transform then two recast transforms on x number of files, then combine all x files into a single bundle.  Right now this doesn't seem possible solely with recast.
